File: server/models/flight.py
from sqlalchemy import Column, String, Float, DateTime, Integer
from sqlalchemy.dialects.postgresql import UUID
import uuid
from datetime import datetime
from database import Base

class Flight(Base):
    __tablename__ = "flights"
    
    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    airline = Column(String, nullable=False)
    departure_city = Column(String, nullable=False)
    destination_city = Column(String, nullable=False)
    departure_time = Column(DateTime, nullable=False)
    arrival_time = Column(DateTime, nullable=False)
    price = Column(Float, nullable=False)
    seats_available = Column(Integer, default=0)
    created_at = Column(DateTime, default=datetime.utcnow)

File: client-next/src/types/flight.ts
export interface Flight {
  id: string;
  airline: string;
  flight_number: string;
  departure_city: string;
  destination_city: string;
  departure_time: string;
  arrival_time: string;
  price: number;
  seats_available: number;
}

export interface FlightListResponse {
  total: number;
  page: number;
  page_size: number;
  results: Flight[];
}